x-pack/plugins/ingest_manager/public/plugin.ts

```diff
@@ -100,27 +100,32 @@ export class IngestManagerPlugin
   }
 
   public async start(core: CoreStart): Promise<IngestManagerStart> {
-    let successPromise: IngestManagerStart['success'];
-    try {
-      const permissionsResponse = await core.http.get<CheckPermissionsResponse>(
-        appRoutesService.getCheckPermissionsPath()
-      );
-
-      if (permissionsResponse?.success) {
-        successPromise = core.http
```

Comment:
Moving from eager/guaranteed to lazy/optional is an important change and we should be on the watch for effects elsewhere.

Until now all the various services could be sure that the plugin setup had at least been started, if not completed, before the other services are used. This meant that some may rely on implied side-effects (default packages, templates, etc) or even race conditions instead of defining those dependencies.

When/if we find them we can add code which uses the global promise or more fine-grained checks, but I don't think we know where those changes will happen right now.

Reply:
Right now I think the only two consumer of this are the ingest manager plugin and endpoint plugin, it should work ( going to ask endpoint to QA that).

Reply:
It's not just about changing the shape returned from start. We are significantly changing the behavior. We cannot know for sure what services depend on setup having been called already, but we do know for sure that it was previously guaranteed to be and now it is not.

It not a deal breaker, but it also not insignificant not easily known (e.g statically analyzable)

Reply:
I agree the change is significant. But also consumer of ingest manager should probably use `isInitialized` to know if the plugin is correctly initialized, this will allow to statically know every consumers.